ray-operator/controllers/ray/rayservice_controller.go

```diff
@@ -576,7 +576,16 @@ func (r *RayServiceReconciler) shouldPrepareNewRayCluster(ctx context.Context, r
 			}
 		}
 
-		// Case 3: Otherwise, rollout a new cluster.
+		// Case 3: If the KubeRay version has changed and the hashes are not identical, update the RayCluster with the goalClusterHash
+		// and new KubeRay version, but do not restart the RayCluster.
+		activeKubeRayVersion := activeRayCluster.ObjectMeta.Annotations[utils.KubeRayVersion]
```

Comment:
How about moving this check to the place between "if activeRayCluster == nil" and "Case 1"?

```go
		if activeRayCluster == nil {
			logger.Info("No active Ray cluster. RayService operator should prepare a new Ray cluster.")
			return RolloutNew
		}
                // Move the check here <--------

		// Case 1: If everything is identical except for the Replicas and WorkersToDelete of
		// each WorkerGroup, then do nothing.
```

Reply:
Done in 5b5fe07